File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    val = -1
    max_val = -1
    min_val = 0
    for r, row in enumerate(sizes):
        for c, val in enumerate(row):
            if val > max_val:
                max_val = val
                min_val = min(row)
                print(max_val, min_val)
                
    for r, row in enumerate(sizes):
        if min(row) > min_val:
            min_val = min(row)
            print(min_val)
            
    answer = max_val * min_val
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> sizes) {
    int answer = 0;
    vector<int> total_min;
    vector<int> total_max;
    for(auto a:sizes){
        int min=a[0];
        int max=a[1];
        if(min>max) {
            min=max;
            max=a[0];
        }
        total_min.push_back(min);
        total_max.push_back(max);
    }
    int total = *max_element(total_min.begin(), total_min.end());
    total *= *max_element(total_max.begin(), total_max.end());
    
    return total;
}